### PR TITLE
Restructure the source files

### PR DESCRIPTION
**Pull request by ephiepark**
This pull request reorganizes codes into different files.
This changes have the following side effect:

- functions like ZSTD_noCompressLiterals, ZSTD_compressRleLiteralsBlock, ZSTD_selectEncodingType, ZSTD_buildCTable, ZSTD_encodeSequences are no longer static functions.

### CODE REVIEW

**Comment by terrelln** [comment]
This is great! `zstd_compress.c` is gigantic

**Comment by Cyan4973** [comment]
The change itself looks good to me.

Now, the main issue is around build scripts.
Some of them will have to be updated, because they can't just "grab the directory content", as `Makefile` does (or are not designed this way, and maybe they should be changed).

This includes :
 - cmake
- Visual Studio
- meson

For the PR to pass, all build systems must be updated.
This is a major source of slow down.

At this stage, a quickfix is good enough.
though if you can find a way to make some of these build systems a bit more "generic", 
this would be a great gain too.

**Comment by ephiepark** [comment]
@Cyan4973  Thanks for the comment! I was confused about what was causing the issue. Let me look into them.

**Comment by ephiepark** [comment]
This pull request had one failure on Travis CI on (Trusty (gcc-8 + ASan + UBSan + Fuzz Test)). This cause of the failure is independent from this pull request. It's caused because gcc8 is not properly suppressing signed-integer-overflow (pointer-overflow) for ubsan. I have created an issue for it (https://github.com/facebook/zstd/issues/1686).

File: build/cmake/lib/CMakeLists.txt [hunk 1]
```diff
@@ -20,62 +20,32 @@ endif()
 # Define library directory, where sources and header files are located
 include_directories(${LIBRARY_DIR} ${LIBRARY_DIR}/common)
 
+file(GLOB CommonSources ${LIBRARY_DIR}/common/*.c)
+file(GLOB CompressSources ${LIBRARY_DIR}/compress/*.c)
+file(GLOB DecompressSources ${LIBRARY_DIR}/decompress/*.c)
+file(GLOB DictBuilderSources ${LIBRARY_DIR}/dictBuilder/*.c)
+file(GLOB DeprecatedSources ${LIBRARY_DIR}/deprecated/*.c)
```
Comment (Cyan4973) [inline build/cmake/lib/CMakeLists.txt]: Thanks @ephiepark , this new `GLOB` is much better for file layout flexibility

**Comment by Cyan4973** [comment]
Thanks @ephiepark ,
this patch is good to go.

As we are close to a new release, we'll wait a bit for the "freeze" snapshot, and merge this patch right after that.

**Comment by Cyan4973** [comment]
Release completed,
we can merge.

**Pull request merged**